packages/js/plugins/ethereum/src/index.ts

```diff
@@ -67,34 +67,39 @@ export class EthereumPlugin extends Module<EthereumPluginConfig> {
     _client: Client
   ): Promise<string> {
     const connection = await this._getConnection(args.connection);
-    const contract = connection.getContract(args.address, [args.method], false);
+    const abi = this._constructAbi(args.method);
+    const contract = connection.getContract(args.address, abi, false);
     const funcs = Object.keys(contract.interface.functions);
-    const res = await contract[funcs[0]](...parseArgs(args.args));
-    return res.toString();
+    const result = await contract[funcs[0]](...parseArgs(args.args));
+    return this._parseResult(abi, result);
   }
 
   async callContractStatic(
     args: Args_callContractStatic,
     _client: Client
   ): Promise<StaticTxResult> {
     const connection = await this._getConnection(args.connection);
-    const contract = connection.getContract(args.address, [args.method]);
+    const abi = this._constructAbi(args.method);
+    const contract = connection.getContract(args.address, abi);
     const funcs = Object.keys(contract.interface.functions);
 
     try {
-      const res = await contract.callStatic[funcs[0]](...parseArgs(args.args), {
-        gasPrice: args.txOverrides?.gasPrice
-          ? ethers.BigNumber.from(args.txOverrides.gasPrice)
-          : undefined,
-        gasLimit: args.txOverrides?.gasLimit
-          ? ethers.BigNumber.from(args.txOverrides.gasLimit)
-          : undefined,
-        value: args.txOverrides?.value
-          ? ethers.BigNumber.from(args.txOverrides.value)
-          : undefined,
-      });
+      const result = await contract.callStatic[funcs[0]](
+        ...parseArgs(args.args),
+        {
+          gasPrice: args.txOverrides?.gasPrice
+            ? ethers.BigNumber.from(args.txOverrides.gasPrice)
+            : undefined,
+          gasLimit: args.txOverrides?.gasLimit
+            ? ethers.BigNumber.from(args.txOverrides.gasLimit)
+            : undefined,
+          value: args.txOverrides?.value
+            ? ethers.BigNumber.from(args.txOverrides.value)
+            : undefined,
+        }
+      );
       return {
-        result: res.toString(),
+        result: result.length ? this._parseResult(abi, result) : "",
```

Comment:
@dOrgJelli im not very proud of this line.. but i wanted to keep compatibility on how it is being formatted now. The alternative was parsing an empty array as <empty string>, and that didnt feel like a right result to return from the `_parseResult` function. 

wyt?

Reply:
What is the type of `result`? Is the `.toString()` in the original code necessary? I feel like you should just be able to do `result: this._parseResult(result)` no?

Reply:
Depending on the function we are calling, `result` will either be an array or an object(string, BigInteger, ..). 
Normally parsing it with the function should be the only thing to do with it, but right now we have a test [here](https://github.com/polywrap/toolchain/blob/origin/packages/js/plugins/ethereum/src/__tests__/e2e.spec.ts#L135) that is asserting that the return value of this function is empty string, and in reality the raw `result` of that function is an empty array(idk why cos the function `function register(bytes32 label, address owner)` doesn't specify any return.. but that comes back).
I fear that changing it could break something if there is someone loosely checking truthy/falsey on the result of staticCall.

But yeah, agree that simply parsing the result is what makes sense.